File: Backend/src/main/java/com/project/backend/services/TransactionService.java
package com.project.backend.services;


import java.time.LocalDate;
import java.util.List;

import com.project.backend.payloads.GraphDataDto;
import com.project.backend.payloads.RecentHistoryResponse;
import com.project.backend.payloads.TotalResponse;

public interface TransactionService {
	List<GraphDataDto> getGraphData(Long userId, LocalDate startDate, LocalDate endDate);
    TotalResponse getTotalIncomeAndExpenses(Long userId);
    RecentHistoryResponse getRecentHistory(Long userId, int limit);
}

File: Frontend/src/services/Transaction-service.js
import { privateAxios } from "./helper";

const TransactionService = {
  getGraphData: (userId, startDate, endDate) => {
    return privateAxios.get(`/user/${userId}/transactions/graph`, {
      params: { startDate, endDate },
    });
  },
  
  getTotalIncomeAndExpenses: (userId) => {
    return privateAxios.get(`/user/${userId}/transactions/total`);
  },
  
  getRecentHistory: (userId, limit = 10) => {
    return privateAxios.get(`/user/${userId}/transactions/recent`, {
      params: { limit },
    });
  },
};

export default TransactionService;
